### PR TITLE
Feature/add composition id input on setting 769

### LINKED ISSUE #769: Add a composition input on the settings page
**Name of feature to enhance**
- Adding a composition id input field on the settings page. 

**Description of feature**
- [ ] Add an input field on the settings page. 
- [ ] Validate the field to make sure it's not empty. 
- [ ] Save the id in the secure shared preferences.

### PR DESCRIPTION
**Pull request by dubdabasoduba**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #769 

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | Code health | Release | Other) 

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/ui/appsetting/AppSettingActivity.kt
Comment (pld) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/ui/appsetting/AppSettingActivity.kt:59]: what is the functional difference btwn the app and composition ID? I assumed these would be equivalent, like if you supplied an app ID, it would know how to map it to a comp ID, why do we also need a comp ID?

Comment (dubdabasoduba) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/ui/appsetting/AppSettingActivity.kt:59]: - This is true we could deprecate the app Id pieces once we fully have the composition id pieces working. Right now we need both for testing purposes until the pieces of the configurable app are fully done.  

- I think we can make this PR with a DO NOT MERGE until we have all the other underlying pieces complete and just couple the work here with deprecating the app id pieces.

**Review by pld — APPROVED** [review]
@dubdabasoduba it's also fine to merge this then remove once we have the other pieces, I'd kinda prefer that so we're all working on the same codebase

**Review by AbdulWahabMemon — APPROVED** [review]
Looks good to me

**Review by pld — APPROVED** [review]
do we want to rebase and merge this now, or circle back after we've merged more config pieces?

**Comment by dubdabasoduba** [comment]
Closing this since it no longer needed.

**Pull request closed without merging**
